components/match2/wikis/trackmania/match_summary.lua

```diff
@@ -0,0 +1,385 @@
+end
+
+function Header:rightOpponent(content)
+	self.rightElement = content
+	return self
+end
+
+function Header:createScoreDisplay(opponent1, opponent2)
+	local function getScore(opponent)
+		local scoreText
+		local isWinner = opponent.placement == 1 or opponent.advances
+		if opponent.placement2 then
+			-- Bracket Reset, show W/L
+			if opponent.placement2 == 1 then
+				isWinner = true
+				scoreText = 'W'
+			else
+				isWinner = false
+				scoreText = 'L'
+			end
+		elseif opponent.extradata and opponent.extradata.additionalScores then
+			-- Match Series (Sets), show the series score
+			scoreText = (opponent.extradata.set1win and 1 or 0)
+					+ (opponent.extradata.set2win and 1 or 0)
+					+ (opponent.extradata.set3win and 1 or 0)
+		else
+			scoreText = OpponentDisplay.InlineScore(opponent)
+		end
+		return OpponentDisplay.BlockScore{
+			isWinner = isWinner,
+			scoreText = scoreText,
+		}
+	end
+
+	return mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:node(
+			getScore(opponent1)
+				:css('margin-right', 0)
+		)
+		:node(' : ')
+		:node(getScore(opponent2))
+end
+
+function Header:createScoreBoard(score, bestof, isNotFinished)
+	local scoreBoardNode = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+
+	if String.isNotEmpty(bestof) and bestof > 0 and isNotFinished then
+		return scoreBoardNode
+			:node(mw.html.create('span')
+				:css('line-height', '1.1')
+				:css('width', '100%')
+				:css('text-align', 'center')
+				:node(score)
+			)
+			:node('<br>')
+			:node(mw.html.create('span')
+				:wikitext('(')
+				:node(Abbreviation.make(
+					'Bo' .. bestof,
+					'Best of ' .. bestof
+				))
+				:wikitext(')')
+			)
+	end
+
+	return scoreBoardNode:node(score)
+end
+
+function Header:soloOpponentTeam(opponent, date)
+	if opponent.type == 'solo' then
+		local teamExists = mw.ext.TeamTemplate.teamexists(opponent.template or '')
+		local display = teamExists
+			and mw.ext.TeamTemplate.teamicon(opponent.template, date)
+			or _TBD_ICON
+		return mw.html.create('div'):wikitext(display)
+			:addClass('brkts-popup-header-opponent-solo-team')
+		end
+end
+
+function Header:createOpponent(opponent, opponentIndex)
+	return OpponentDisplay.BlockOpponent({
+		flip = opponentIndex == 1,
+		opponent = opponent,
+		overflow = 'ellipsis',
+		teamStyle = 'short',
+	})
+		:addClass(opponent.type ~= 'solo'
+			and 'brkts-popup-header-opponent'
+			or 'brkts-popup-header-opponent-solo-with-team')
+end
+
+function Header:create()
+	self.root:tag('div'):addClass('brkts-popup-header-opponent'):addClass('brkts-popup-header-opponent-left')
+		:node(self.leftElementAdditional)
+		:node(self.leftElement)
+	self.root:node(self.scoreBoard)
+	self.root:tag('div'):addClass('brkts-popup-header-opponent'):addClass('brkts-popup-header-opponent-right')
+		:node(self.rightElement)
+		:node(self.rightElementAdditional)
+	return self.root
+end
+
+
+local CustomMatchSummary = {}
+
+function CustomMatchSummary._getHeadToHead(opponents)
```

Comment:
needed?

Reply:
It could be useful because it is one of the tiebreakers involved in the rulebook

Reply:
i think you misunderstand what the function does
it creates an icon in the footer of the matchsummary that links to h2h pages between the participants 
it is only useful if the h2h stuff (a form/special:RunQuery page that invokes a module that does query for matches between the entered participants) is set up on the wiki

Reply:
Yup I understood this, the thing is we did not made the h2h module yet, but I think it will be useful once we've done it